### PR TITLE
Add Infinite Loop Research prompt

### PR DESCRIPTION
**Pull request by chownces**
### Description

~***NOTE: Please do not merge this till the main Rook PR has been reviewed and merged***~

Adds a prompt to seek the user's consent for participation in research. This prompt appears for every new course a user creates/ joins.

~*Do update the prompt message before merging!*~

image

### Type of change

- [x] New feature (non-breaking change which adds functionality)

### How to test

To test with the backend PR https://github.com/source-academy/backend/pull/790

### Checklist

- [x] I have tested this code

### CODE REVIEW

File: src/commons/application/Application.tsx [hunk 1]
```diff
@@ -237,6 +240,49 @@ const Application: React.FC<ApplicationProps> = props => {
           </Switch>
         )}
       </div>
+
+      {/* agreedToResearch has a default value of undefined in the store.
+          It will take on null/true/false when the backend returns. */}
+      {props.agreedToResearch === null && (
+        <div className="research-prompt">
+          <Dialog
```
Comment (chownces) [inline src/commons/application/Application.tsx:248]: Im not sure if this is the best way to handle this. This PR (and the backend https://github.com/source-academy/backend/pull/790) does not take into account instructors who run their own Source Academy course on their own backend. In such instances, the prompt should not display (and ideally the backend should not track or send this boolean either)

Comment (chownces) [reply src/commons/application/Application.tsx:248]: another env variable in the frontend?

Comment (angelsl) [reply src/commons/application/Application.tsx:248]: @chownces Yeah, should add a flag for this.

**Review by angelsl — APPROVED** [review]
Looks good, but do add a toggle for this (as an env. var, default off) first

**Review by angelsl — APPROVED** [review]

**Pull request merged**